### ISSUE TITLE
Godot runs slower with GLES2 than GLES3

### ISSUE THREAD

**Issue by sdfgeoff** [godotengine/godot#24522]
**Godot version:**
Recent master: 
26058376aa180f70e152070b3b69272287a710a8

**OS/device including version:**
OS:  Ubuntu 18.04 with many things removed for performance
Hardware: Intel Atom x5-8350, 2Gb ram. Integrated GPU ("HD Graphics (Cherry Trail)")

**Issue description:**
Yes, I'm aware that this is a very low spec piece of hardware. I was not expecting it to run well, and actually, the fact that Godot ran at all was a pleasant surprise. The other surprise was:

- When running with `--video-driver GLES3 --print-fps` the result is around 34 FPS
- When running with `--video-driver GLES2 --print-fps` the result is around 18 FPS

Description of the scene:
 - One directional lamp with shadow
 - About 10-20 objects, all with "normal" spatial shaders. Two materials have a single texture.
 - A GUI

I'm not sure if this is an issue, or if it's just myself misunderstanding the purpose of GLES2 mode, so:
_Should GLES2 have better performance than GLES3_

**Comment by volzhs** [comment]
#24466 pointed it too

**Comment by sdfgeoff** [comment]
No difference with 3c999ae1a1f897716041b45606a4506790da1e91 FPS is still slower on GLES2

**Comment by OBKF** [comment]
I also have noticed a drop in FPS and a high CPU usage when using the GLES2 (not always) compared to GLES3. Godot2 was way too fast for 3D, I thought it was because Godot3's GLES2 looks better.
I also have noticed that using multithreading causes more lag even if you get 10-40 FPS.

**Comment by avril-gh** [comment]
on this 1039ba9 and before, i seen 50-55 FPS on android GLES2 devices,
after this #24445 has been fixed i noticed FPS has drop to 9-10 FPS.

**Comment by sdfgeoff** [comment]
Um, this is a single directional lamp with a shadow. I do not agree that that should run slower

**Comment by akien-mga** [comment]
For the reference, based on my tests in https://github.com/godotengine/godot/issues/24466#issuecomment-454078286 and once #24987 is merged, GLES2 runs faster than GLES3 again (in simple scenarios at least).

**Comment by segamanov** [comment]
Guys. Help pliz. I have on gles3 albedo strange displays. On gles2 it seems normal. This is only on the node Shader. On the usual - all is well. What might be the problem? I have linux mint.

**Comment by Calinou** [comment]
@segamanov Please open a new issue for this. Make sure to attach a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) along the way.

**Comment by segamanov** [comment]
My English is bad. Even worse I know your site)) Unfortunately, I deleted everything...11.08.2020, 19:19, "Hugo Locurcio" <notifications@github.com>:
@segamanov Please open a new issue for this. Make sure to attach a minimal reproduction project along the way.
